src/prefect/agent.py

```diff
@@ -425,6 +425,16 @@ async def get_infrastructure(self, flow_run: FlowRun) -> Infrastructure:
             # once we reach the end, set the value
             data[nested_fields[-1]] = value
 
+        # FlowRun's infra_overrides should take precedence over Deployment's
```

Comment:
I think this should be the only thing to change in order to make `FlowRun`'s `infra_overrides` override `Deployment`'s, right?